### ISSUE TITLE
custom class instance can't access members after being loaded from PackedScene

### ISSUE THREAD

**Issue by KiwwiPity** [godotengine/godot#61947]
### Godot version

3.4.3 stable

### System information

Linux Debian 11

### Issue description

When a custom class instance whose `_init()` expects a parameter, and is stored on disk in a `PackedScene`, to then load it and access the members of such instance, it isn't recognized as custom class, only as `Node`, even if extends from another built-in type (in this case `Node2D`), so can't access its members.

When try it, the compiler complains.

Play with the exported `bool` property in the main scene.

### Steps to reproduce

The adjunct project has all to reproduce it and has comments for guidance.

Press `F5` to reproduce the issue.

### Minimal reproduction project

[custom_class_instance_cant_access_members.zip](https://github.com/godotengine/godot/files/8884797/custom_class_instance_cant_access_members.zip)

**Comment by Calinou** [comment]
> When a custom class instance whose _init() expects a parameter

Custom nodes should not expect parameters in their constructor, or they won't be addable as nodes anymore to the scene.

**Comment by KiwwiPity** [comment]
> > When a custom class instance whose `_init()` expects a parameter
> 
> The node's script fails to initialize in this case. Custom nodes should not expect parameters in their `_init()` constructor, or they won't be addable as nodes anymore to the scene.

That sounds rare for me, as it is the constructor of GDScript.
Then it means that must use another kind of ctor, or must initialize instances by accessing directly its members?

**Comment by Calinou** [comment]
> Then it means that must use another kind of ctor, or must initialize instances by accessing directly its members?

Indeed, you need to access instances' members to initialize them externally.

**Comment by HolonProduction** [comment]
Closing this. The fact that serialized nodes and resources need a no-arg constructor is known and documented. It's not a bug, and improvements to it need an actionable proposal.
